rfcs/engine-developer-guide.md

```diff
@@ -0,0 +1,127 @@
+## New guide: Engine Developer Guide
+
+Currently, users of Open 3D Engine (O3DE) only have one primary product manual: The User Guide (we also offer the Atom Guide, which is intended to be rolled into the User Guide at a future date.) While for most project developers and content creators the user guide is intended to be sufficient, O3DE has other audiences:
+
+* O3DE is an open-source product that is always actively seeking contributors onboarding into its code base.
+* O3DE operates in a product space where developers often need very specific low-level technical details, or guidance on performing custom implementations of the core engine.
+
+To support those considerations, there should be an Engine Developer Guide added to o3de.org. The Engine Developer Guide would cover low-level documentation for contributors and engine modification from an official, vetted, quality-controlled source.
+Motivation
+
+Contributors and developers modifying the engine require low-level documentation to perform their tasks. Some existing resources that they have access to which can provide some of this information are: Source code; Header documentation and API reference; RFCs.
+
+While new features in the engine require RFCs, this information is not available for all parts of O3DE. In addition, header documentation and API reference have many issues in the project's current state. In particular, API reference has been a lower priority for developers. Relying on source code can be difficult for even advanced users, especially in cases where there are many abstraction layers. With the modular design of O3DE, this is a particular concern.
+
+While there are long-term solutions to some of these issues such as API reference requirements for source contribution, we need a more immediate stopgap that allows developers to share notes, reference, and other critical implementation documentation. The Engine Developer Guide is intended to serve this purpose, and carrying forward, provide authoritative information on designs and implementation.
+
+### Advantages
+
+* Provides a dedicated repository of information for low-level developers, available from the same source as user-facing documentation. Initiatives to expand existing API reference and core documentation in the future will help focus the contents of the engine developer guide.
```

Comment:
Do we have a definition of "low-level developers"?

Reply:
Covered on Line 8:
> The Engine Developer Guide would cover low-level documentation for contributors and engine modification from an official, vetted, quality-controlled source.